app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -46,6 +48,19 @@ def create_sub_plans
         end
       end
 
+      def spawn_plans
+        super
+        if output[:planned_count] - output[:remote_triggered_count] >= proxy_batch_size
+          plan_event(Actions::TriggerProxyBatch::TriggerLastBatch, nil, step_id: input[:trigger_run_step_id])
```

Comment:
Is this necessary? Won't it lead to TriggerLastBatch being dispatched twice?

Reply:
That's outside wrong, this should be `TriggerNextBatch`